vyper/functions/functions.py

```diff
@@ -616,7 +616,7 @@ def as_wei_value(expr, args, kwargs, context):
         if value.value <= 0:
             raise InvalidLiteralException("Negative wei value not allowed", expr)
         sub = ['mul', value.value, denom_divisor]
-    elif value.typ.typ == 'uint256':
+    elif value.typ.typ in ['uint256', 'int128']:
         sub = ['mul', value, denom_divisor]
     else:
```

Comment:
as mentioned in #1637 this should probably be `elif value.typ.typ == 'decimal'`, and then add a catch-all else branch which throws some sort of 'type not found' exception